frontend/packages/ceph-storage-plugin/src/components/bucket-class/shared.ts

```diff
@@ -0,0 +1,54 @@
+const bucketClassNameRegex: RegExp = /^[a-z0-9]+[a-z0-9-]+[a-z0-9]+$/;
+export enum BucketClassTypes {
+  STANDARD = 'Standard',
+  NAMESPACE = 'Namespace',
+}
+export const bucketClassTypeRadios = [
+  {
+    id: BucketClassTypes.STANDARD,
+    value: BucketClassTypes.STANDARD,
+    label: 'Standard',
+    description:
+      'Data will be ingested by Multi-cloud object gateway, dedeuped, compressed and encrypted. The encrypted chunks would be saved on the selected backing stores. Best used when the applications would always use the OpenShift Container Storage endpoints to access the data.',
+  },
+  {
+    id: BucketClassTypes.NAMESPACE,
+    value: BucketClassTypes.NAMESPACE,
+    label: 'Namespace',
+    description:
+      'Data will be stored as is(no dedup, compression, encryption) on the namespace stores. Namespace buckets allows for connecting to existing data and serving from them. Best used for existing data or when other applications (and native cloud services) need to access the data from outside the OpenShift Container Storage.',
+  },
+];
+
+export const validateBucketClassName = (name: string): boolean => {
+  return name.length >= 3 && name.length <= 63 && bucketClassNameRegex.test(name);
+};
+
+export enum NamespacePolicyType {
+  SINGLE = 'Single',
+  MULTI = 'Multi',
+  CACHE = 'Cache',
+}
+
+export const namespacePolicyTypeRadios = [
+  {
+    id: NamespacePolicyType.SINGLE,
+    value: NamespacePolicyType.SINGLE,
+    label: 'Single namespace-store',
+    description: 'The namespace bucket will read and write its data to a selected namespace store',
+  },
+  {
+    id: NamespacePolicyType.MULTI,
+    value: NamespacePolicyType.MULTI,
+    label: 'Multi namespace-stores',
+    description:
+      'The namespace bucket will serve reads from several selected backing stores, creating a virtual namespace on top of them and will write to one of those as its chosen write target',
+  },
+  {
+    id: NamespacePolicyType.CACHE,
+    value: NamespacePolicyType.CACHE,
+    label: 'Cache namespace-store',
+    description:
+      'The caching bucket will serve data from a large raw data out of a local caching tiering.',
+  },
+];
```

Comment:
These can be part of `constants/bucket-class`

Reply:
changed